File: lab4/src/warehouse/resources/Ammunition.java
package warehouse.resources;

import enums.Caliber;
import enums.MeasureUnit;
import enums.SupplyType;
import request.SupplyRequestDetail;

public class Ammunition extends Resource{
    Caliber caliber;
    public Ammunition(int id, String name, Caliber caliber, int count)
    {
        super(id, name, MeasureUnit.ITEM, count);
        this.caliber = caliber;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.AMMUNITION &&
               detail.getCaliber() == caliber;
    }

    public Caliber getCaliber() {
        return caliber;
    }
}

File: lab5/C++/src/Ammunition.cpp
#include "Ammunition.hpp"
#include "SupplyRequestDetail.hpp"

Ammunition::Ammunition(const Ammunition& other) :
    Resource(rand(), other.GetName(), MeasureUnit::eItem, other.GetQuantity()), m_caliber(other.m_caliber) { }

Ammunition::Ammunition(int id, const std::string& name, Caliber caliber, float quantity)
    : Resource(id, name, MeasureUnit::eItem, quantity), m_caliber(caliber) {
}

bool Ammunition::IsMatches(const SupplyRequestDetail& detail) const {
    return detail.GetSupplyType() == SupplyType::Ammunition && detail.GetCaliber() == m_caliber;
}

Caliber Ammunition::GetCaliber() const {
    return m_caliber;
}